front/lib/resources/agent_message_feedback_resource.ts

```diff
@@ -43,130 +61,357 @@ export class AgentMessageFeedbackResource extends BaseResource<AgentMessageFeedb
+    filters?: {
+      limit?: number;
+      olderThan?: Date;
+      earlierThan?: Date;
+    };
+  }): Promise<
+    (AgentMessageFeedbackType | AgentMessageFeedbackWithMetadataType)[]
+  > {
+    const createdAtClause =
+      !filters || (!filters.olderThan && !filters.earlierThan)
+        ? {}
+        : {
+            createdAt: {
+              [Op.and]: [
+                filters.olderThan && filters.earlierThan
+                  ? [
+                      { [Op.lt]: filters.olderThan },
+                      { [Op.gt]: filters.earlierThan },
+                    ]
+                  : filters.olderThan
+                    ? { [Op.lt]: filters.olderThan }
+                    : { [Op.gt]: filters.earlierThan },
+              ],
+            },
+          };
+
+    const agentMessageFeedback = await AgentMessageFeedback.findAll({
       where: {
-        userId: user.id,
-        agentMessageId: agentMessage.id,
+        // Necessary for global models who share ids across workspaces
+        workspaceId: workspaceId.toString(),
+        // These clauses are optional
+        ...(agentConfiguration
+          ? {
+              agentConfigurationId: agentConfiguration.sId.toString(),
+            }
+          : {}),
+        ...createdAtClause,
       },
+
+      include: [
+        {
+          model: AgentMessageModel,
+          attributes: ["id"],
+          as: "agentMessage",
+          include: [
+            {
+              model: Message,
+              as: "message",
+              attributes: ["id", "sId"],
+              include: [
+                {
+                  model: Conversation,
+                  as: "conversation",
+                  attributes: ["id", "sId"],
+                },
+              ],
+            },
+          ],
+        },
+        {
+          model: UserResource.model,
+          attributes: ["name", "imageUrl", "email"],
+        },
+      ],
+      order: [
+        ["agentConfigurationVersion", "DESC"],
+        ["createdAt", "DESC"],
+      ],
+      limit: filters?.limit,
     });
 
-    if (!agentMessageFeedback) {
-      return null;
-    }
+    return (
+      agentMessageFeedback
+        // Typeguard needed because of TypeScript limitations
+        .filter(
+          (
+            feedback
+          ): feedback is AgentMessageFeedback & {
+            agentMessage: { message: Message & { conversation: Conversation } };
+          } => !!feedback.agentMessage?.message?.conversation
+        )
+        .map((feedback) => {
+          return {
+            id: feedback.id,
+            messageId: feedback.agentMessage.message.sId,
+            agentMessageId: feedback.agentMessageId,
+            userId: feedback.userId,
+            thumbDirection: feedback.thumbDirection,
+            content: feedback.content
+              ? feedback.content.replace(/\r?\n/g, "\\n")
+              : null,
+            isConversationShared: feedback.isConversationShared,
```

Comment:
This is repetitive of the presence of conversationId right? Do we really need to expose it on AgentMessageFeedbackType

Reply:
I like to expose it, so that this 'hiding' of the conversationId becomes a nice to have but not a logic we propagate and rely on.
This way we can have feedbacks-related functions include extra checks on isConversationShared rather than just assuming that the conversationId is already secured.